addon/services/breadcrumbs.js

```diff
@@ -15,6 +15,42 @@ export default Service.extend({
   init () {
     this._super(...arguments)
 
-    this.instances = A([])
+    this.containers = []
+  },
+
+  /**
+   * Methods
+   */
+
+  registerContainer (containerData) {
+    assert(
+      'An `element` is required to register a breadcrumb container',
+      containerData.element
+    )
+    assert(
+      'This container was already registered before',
+      !this.isContainerRegistered(containerData)
+    )
+
+    this.set('containers', [...this.containers, containerData])
+  },
+
+  unregisterContainer ({ element }) {
+    this.set(
+      'containers',
+      this.containers.filter(registeredContainer => {
+        return element !== registeredContainer.element
+      })
+    )
+  },
+
+  /* Private */
```

Comment:
should we just start prefixing with `_` like most js devs?

Reply:
Not sure, could do, slightly indifferent about it.

Reply:
I hate it, but it does make it obvious which things should not be touched from the outside. I just which JS came up with native private scoping (without h4x).

Reply:
Yup, underscoring is fine for me as well.